src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import command.*;
+import exception.UnknownInputException;
+import exception.EmptyDescriptionException;
+
+/**
+ * Represents a parser object to parse input commands
+ */
+public class Parser {
+    private DukeList dukelist;
+    private Storage storage;
+
+    /**
+     * Creates a Parser object
+     *
+     * @param dukeList a DukeList that can be accessed by this Parser
+     * @param storage a Storage that is used to read and write
+     */
+    public Parser (DukeList dukeList, Storage storage ) {
+        this.dukelist = dukeList;
+        this.storage = storage;
+    }
+
+    /**
+     * Translates an input String to its corresponding command
+     * @param inputText the input String that is to be parsed
+     * @return the command that corresponds to the input text
+     * @throws EmptyDescriptionException when there is a valid command with no content
+     * @throws UnknownInputException when the input text doe not correspond to any known command
+     */
+    public Command parse(String inputText) throws EmptyDescriptionException, UnknownInputException{
+        String[] array = inputText.split(" ", 2);
+        try {
+            String first = array[0];
+            if (inputText.equals("bye")) {
+                return new ExitCommand(this.dukelist, this.storage);
+            } else if (first.equals("list")){
+                return new ListCommand(this.dukelist);
+            } else if (first.equals("mark")) {
+                return new MarkCommand(array[1], true, dukelist);
+            } else if (first.equals("unmark")) {
+                return new MarkCommand(array[1], false, dukelist);
+            } else if (first.equals("delete")){
+                return new DeleteCommand(array[1], dukelist);
+            } else if (first.equals("find")) {
+                return new FindCommand(array[1], dukelist);
+            } else {
+                if (array.length != 2) {
+                    if (first.equals("todo") || first.equals("deadline") || first.equals("event")) {
```

Comment:
Maybe u can consider using switch for this part so it looks cleaner and improve readability?